operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.operator.crds.v2alpha1.deployment.spec;
+
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+import io.fabric8.kubernetes.api.model.SecretKeySelector;
+import io.sundr.builder.annotations.Buildable;
+
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class DatabaseSpec {
+
+    @JsonPropertyDescription("The database vendor.")
+    private String vendor;
+
+    @JsonPropertyDescription("The reference to a secret holding the username of the database user.")
+    private SecretKeySelector usernameSecret;
+
+    @JsonPropertyDescription("The reference to a secret holding the password of the database user.")
+    private SecretKeySelector passwordSecret;
+
+    @JsonPropertyDescription("Sets the database name of the default JDBC URL of the chosen vendor. If the `url` option is set, this option is ignored.")
+    private String database;
+
+    @JsonPropertyDescription("Sets the hostname of the default JDBC URL of the chosen vendor. If the `url` option is set, this option is ignored.")
+    private String host;
+
+    @JsonPropertyDescription("Sets the port of the default JDBC URL of the chosen vendor. If the `url` option is set, this option is ignored.")
+    private Integer port;
+
+    @JsonPropertyDescription("The database schema to be used.")
+    private String schema;
+
+    @JsonPropertyDescription("The full database JDBC URL. If not provided, a default URL is set based on the selected database vendor. " +
+            "For instance, if using 'postgres', the default JDBC URL would be 'jdbc:postgresql://localhost/keycloak'. ")
+    private String url;
+
+    @JsonPropertyDescription("The initial size of the connection pool.")
+    private Integer poolInitialSize;
+
+    @JsonPropertyDescription("The minimal size of the connection pool.")
+    private Integer poolMinSize;
+
+    @JsonPropertyDescription("The maximum size of the connection pool.")
+    private Integer poolMaxSize;
+
```

Comment:
So, the `db-url-properties` will be omitted for now?

Reply:
I think we should and if needed, people can use the `url` option.

Reply:
ok, agree